subsys/logging/log_core.c

```diff
@@ -375,7 +386,9 @@ static u32_t count_args(const char *fmt)
 
 void log_generic(struct log_msg_ids src_level, const char *fmt, va_list ap)
 {
-	if (IS_ENABLED(CONFIG_LOG_IMMEDIATE) &&
+	if (_is_user_context()) {
```

Comment:
same as above.

I assume you rely on the compiler to automatically remove these, just wondering if you've checked

Reply:
Addressed above.